cirq/experiments/qubit_characterizations_test.py

```diff
@@ -40,6 +42,37 @@ def test_rabi_oscillations():
     assert rms_err < 0.1
 
 
+def test_single_qubit_cliffords():
```

Comment:
Optional: It seems easy to write a unit test that verifies the key property of the Clifford group - that it fixes the Pauli group - but it seems we don't have it. WDYT about adding it?

(Strictly speaking this isn't related to this PR, but if we had this test then observing it pass before and after would provide a lot of confidence in the new logic.)

Reply:
I agree we should have some better abstractions around the pauli group and clifford group, but I think I'll leave that as future work because we'll probably need some design discussion around that.